wcfsetup/install/files/lib/form/RegisterForm.class.php

```diff
@@ -403,6 +404,7 @@ public function save() {
 			// create fake password
 			$this->password = StringUtil::getRandomID();
 		}
+		EventHandler::getInstance()->fireAction($registerVia3rdParty, 'registerVia3rdParty');
```

Comment:
This does not work like this. You need to specify a valid object or class name as the first parameter. On top of that the only parameter that is passed by reference is the third one. Pass an array as the third parameter and after the event update the variable with the current contents of that array.

While doing so also have a look whether there is anything else you'd like to modify, possibly creating a more generally useful event.